### PR TITLE
Add new IcoswISC30E3r5 ocean and sea-ice mesh

### PR DESCRIPTION
**Pull request by xylar**
Long name: IcoswISC30L64E3SMv3r5

This nearly uniform 30 km mesh is the dual mesh of a subdivided icosahedron (Icos).  The mesh is with Ice-Shelf Cavities (wISC) around Antarctica and has 64 vertical layers. Unlike [IcoswISC30E3r4](https://github.com/E3SM-Project/E3SM/pull/6078), this version of the mesh **does not** include smoothing of topography.

This mesh is a candidate for the E3SM v3 (E3) low res mesh.  It is revision 5 (r5) of the mesh.

The mesh was created using [compass](https://github.com/MPAS-Dev/compass), and is being reviewed there in this PR:
https://github.com/MPAS-Dev/compass/pull/735
As soon as the PR is merged, the code used to produce it will be at this tag: 
https://github.com/MPAS-Dev/compass/releases/tag/mesh_IcoswISC30E3r5

The mesh and simulation results (to follow) are been reviewed and approved here:
https://acme-climate.atlassian.net/wiki/spaces/OO/pages/4006641665/Review+IcoswISC30E3r5

A B-case will begin shortly and analysis will be posted here and on the review page as soon as it is available.

### CODE REVIEW

**Comment by xylar** [comment]
A screenshot of the unsmoothed bathymetry:
bathy_unsmooth
Compare with:
https://github.com/E3SM-Project/E3SM/pull/6078#issuecomment-1819989371

File: cime_config/config_grids.xml [hunk 1]
```diff
@@ -2771,6 +2828,13 @@
       <desc>ECwISC30to60E3r2 is a MPAS ocean grid generated with the jigsaw/compass process using the eddy closure density function that has 30 km gridcells at the equator, 60 km at mid-latitudes, and 35 km at high latitudes. Additionally, it has ocean in ice-shelf cavities:</desc>
     </domain>
 
+    <domain name="IcoswISC30E3r5">
+      <nx>465044</nx>
+      <ny>1</ny>
+      <file grid="ice|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.IcoswISC30E3r5.231121.nc</file>
+      <desc>IcoswISC30E3r5 is a MPAS ocean grid generated with the jigsaw/compass process XXX. Additionally, it has ocean in ice-shelf cavities:</desc>
```
Comment (rljacob) [inline cime_config/config_grids.xml:2835]: Should "XXX" be replaced with something?

**Review by rljacob — APPROVED** [review]
Map names look good!

File: cime_config/config_grids.xml (hunk 1 above)
Comment (jonbob) [reply cime_config/config_grids.xml:2835]: Thanks for the reminder @rljacob. I'll pull the description from the PR that was closed and add it here

File: cime_config/config_grids.xml [hunk 2]
```diff
@@ -3976,12 +4058,12 @@
...
-      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/T62/map_T62_to_ECwISC30to60E3r2-nomask_trintbilin.20231018.nc</map>
-      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/T62/map_T62_to_ECwISC30to60E3r2_patch.20231018.nc</map>
-      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/ECwISC30to60E3r2/map_ECwISC30to60E3r2_to_T62_traave.20231018.nc</map>
-      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/ECwISC30to60E3r2/map_ECwISC30to60E3r2_to_T62_traave.20231018.nc</map>
+    <gridmap atm_grid="T62" ocn_grid="IcoswISC30E3r5">
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/T62/map_T62_to_IcoswISC30E3r5_traave.20231121.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/T62/map_T62_to_IcoswISC30E3r5-nomask_trbilin.20231121.nc</map>
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/T62/map_T62_to_IcoswISC30E3r5_esmfpatch.20231121.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_T62_traave.20231121.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/IcoswISC30E3r5/map_IcoswISC30E3r5_to_T62_traave.20231121.nc</map>
```
Comment (xylar) [inline cime_config/config_grids.xml:4066]: @jonbob, similar question as in #6078, was this removed on purpose?

**Comment by xylar** [comment]
I rebased to handle conflicts with AB2 merge, #5989

**Comment by xylar** [comment]
@vanroekel and @proteanplanet, as you heard this morning, @golaz would like to have this merged to `master` before the control run begins.  Please let me know if there is something you need from me before you are ready to review and hopefully approve this PR.

File: cime_config/config_grids.xml (hunk 2 above)
Comment (xylar) [reply cime_config/config_grids.xml]: Fixed!

**Review by proteanplanet — APPROVED** [review]
I have reviewed this mesh visually and numerically, checking for narrow ocean passages, coastal bathymetry, and ice shelf cavity correctness.  The mesh largely mimics IcoswISC30E3r2 and therefore I won't post the analysis again because it's almost identical to that mesh, except for the important updates that are made in this PR.  Approved.

**Review by mark-petersen — APPROVED** [review]
I added images of bathymetry and coastlines to https://acme-climate.atlassian.net/wiki/spaces/OO/pages/4006641665/Review+IcoswISC30E3r5 and verified that critical passages are open, and that these fields look correct.

**Review by vanroekel — APPROVED** [review]
approving by visual inspection and examination of analysis from @proteanplanet and @mark-petersen

**Comment by jonbob** [comment]
passes sanity testing, merged to next

**Comment by jonbob** [comment]
merged to master -- and double-checked that all required files are in the inputdata repo

**Comment by xylar** [comment]
Woot! Woot!  I'll copy over any remaining diagnostics files, etc. from the compass run to `public_html`.

**Comment by xylar** [comment]
Done!

**Pull request merged**